### ISSUE TITLE
Nextcloud credentials don't save

### ISSUE THREAD

**Issue by Gaspartame** [quillpad/quillpad#168]
When trying to setup a sync with my nextcloud notes, after filling `Nextcloud instance URL`, I fill `Nextcloud account` with my credentials then tap on `save`, but nothing happens.
version `1.4.10` from f.droid on android 10

**Comment by Gaspartame** [comment]
Maybe I should use the CalDav url instead of the Nextcloud url as I read you avoided treating with the Nextcloud API ?

**Comment by EphemeralDev** [comment]
Quillpad connects to the notes app in nextcloud, see #166.

**Comment by Gaspartame** [comment]
Sorry by *app* I meant app on the phone. Quillpad doesn't require the Nextcloud app to be installed on the phone and logged to a server account does it ?

**Comment by EphemeralDev** [comment]
I think you need to install/add the Notes app (in your nextcloud instance) and ensure the Notes api is exposed on your server. Sorry, i'm not familiar with nextcloud; I'm unsure if you need the Nextcloud app itself installed.

**Comment by Gaspartame** [comment]
The Notes app is well installed and API exposed because I use another phone app that works with the Note API on the same instance.
I think my problem is more related to a UI bug. Note that I'm running android 11 in Sailfish Os, but that shouldn't change anything for the app. Meanwhile, I can surely do some debugging if you're interested.

**Comment by jsixface** [comment]
Is your nextcloud url a subdomain or behind a root  path?

**Comment by EmJotGeh** [comment]
Same problem here:

after reinstalling and first start the credentials won't save. Some try's later the URL now saved but when i try to use my login credentials i got every time "Es ist ein Problem aufgetreten" 
Quillpad worked perfectly before I had to reinstall my smartphone.

**Comment by jsixface** [comment]
Can you try installing the previous version from either GitHub release page or f-droid?
If it works on 1.4.9, I can compare the changes.

**Comment by christheday** [comment]
I had the same issue with 1.4.10.  I just installed QuillPad and tried to connect to my nextcloud server.  I downgraded to 1.4.9 and saw the same issue.
Does QuillPad not support self-signed certificates? My nextcloud server is on my home LAN and is accessed using tailscale.

**Comment by jsixface** [comment]
No, sorry. Self signed SSL certificates does not work. Does it work on the Nextcloud app on your phone if you installed it?

**Comment by jsixface** [comment]
@Gaspartame Do you use a self signed certificate too?

**Comment by christheday** [comment]
@jsixface Yes, the Nextcloud app lets you trust a self-signed certificate. The Nextcloud News app allows the same thing and allows me to log in using the standard Nextcloud app as well.

https://github.com/nextcloud/news

**Comment by EmJotGeh** [comment]
I tried it with previous versions but the problem persists. I use a Let'sEncrypt certificate.

**Comment by Motor-head** [comment]
> No, sorry. Self signed SSL certificates does not work. Does it work on the Nextcloud app on your phone if you installed it?

@jsixface any reason for self signed SSL certificates to not work?

**Comment by jsixface** [comment]
Technically we can make it work. The app in its current state cannot verify the issuer of those certificates and that's why it fails. We should be able to provide a checkbox to make sure the user is aware of the self-signed certificate and make the app accept that.

**Comment by Silejonu** [comment]
Can't Quillpad trust the certificates in Android's trust store? My certificate authority is added in Android, and all apps trust my issued certificates (Firefox, NextCloud…). Does Quillpad currently use its own trust store?

**Comment by nodiscc** [comment]
Currently facing the same problem
- Nextcloud instance using a self-signed certificate
- Android 10
- self-signed certificate installed in Android trust store
- Logging in to Nextcloud from Quillpad fails with `trust anchor for certification path not found`

> Can't Quillpad trust the certificates in Android's trust store?

At least for me it does not.

**Comment by oyvigri** [comment]
Same. 
Not having the self signed SSL cert in trust store give me the Java certpath etc exception, and having the cert in the trust store givese same as @nodiscc.
There's always the possibility ife doing something wrong ofc. If not it would really be amazing having this fixed
